config/config.php

```diff
@@ -36,11 +36,22 @@
 /**
  * Hooks
  */
-$GLOBALS['TL_HOOKS']['parseArticles'][] = array('News', 'addCategoriesToTemplate');
-$GLOBALS['TL_HOOKS']['replaceInsertTags'][] = array('News', 'parseCategoriesTags');
+$GLOBALS['TL_HOOKS']['parseArticles']['news_categories']     = ['News', 'addCategoriesToTemplate'];
+$GLOBALS['TL_HOOKS']['replaceInsertTags']['news_categories'] = ['News', 'parseCategoriesTags'];
+$GLOBALS['TL_HOOKS']['replaceInsertTags']['news_categories'] = ['InsertTags', 'replace'];
 
-if (in_array('changelanguage', \ModuleLoader::getActive())) {
-    $GLOBALS['TL_HOOKS']['translateUrlParameters'][] = array('NewsCategories', 'translateUrlParameters');
+
+if (in_array('changelanguage', \ModuleLoader::getActive()))
+{
+    $GLOBALS['TL_HOOKS']['translateUrlParameters'][] = ['NewsCategories', 'translateUrlParameters'];
+}
+
+foreach ($GLOBALS['TL_HOOKS']['getSearchablePages'] as &$arrCallback)
+{
+    if ($arrCallback[0] == 'News' && $arrCallback[1] == 'getSearchablePages')
+    {
+        $arrCallback = ['News', 'getSearchablePages'];
```

Comment:
Huh? 😄

Reply:
Changed it to :+1: 

```
foreach ($GLOBALS['TL_HOOKS']['getSearchablePages'] as &$arrCallback)
{
    if ($arrCallback[0] == 'News' && $arrCallback[1] == 'getSearchablePages')
    {
        $arrCallback = ['NewsCategories\News', 'getSearchablePages'];
    }
}
```
No idea why it worked already within wrong namespace, but now it more clear. News::getLink() can now be overwritten within `NewsCategories\News`.

Reply:
It worked because we register the `News` class using the default Contao class autoloader which in fact overwrites the class reference in the global namespace, so in fact you could completely drop this hooks update and it should work too. What Contao autoloader does is completely wrong but let's leave it for now...

Reply:
Thanks for your clarification. Better keep this, if contao switch from its autoloader? You are right, if i remove this code, it still works.